### PR TITLE
Use Web Workers for background processing

### PR DESCRIPTION
**Pull request by Uberi**
* Web Workers allow us to do the heavy lifting off of the UI thread, which keeps the window responsive.
* There is still a little bit of stutter when the charts are being drawn, which is unavoidable AFAIK.
* This adds a file to the Telemetry distribution called `telemetry-worker.js`, the background worker, as well as functions like `Telemetry.doAsync` for callback-style async worker operations.
* The worker is only used for histogram precomputation and entry counting, since those were the only real hotspots.
* A small optimization to bucketing also caused a gigantic speedup in histogram evolution creation.

### CODE REVIEW

**Comment by Uberi** [comment]
This is now live at http://performer-chimpanzee-56033.bitballoon.com/ for testing purposes.

File: v1/telemetry-worker.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
+importScripts("telemetry.js", "big.js")
+
+var gActions = {
+  "Histogram-count": function() {
```
Comment (vdjeric) [inline v1/telemetry-worker.js]: use the same style as the other files, we don't quote function names or use hyphens inside the names

File: v1/telemetry.js [hunk 2]
```diff
@@ -1468,8 +1514,8 @@ Histogram.prototype.geometricStandardDeviation = function() {
     );
   }
   var count       = this.count();
-  var log_sum     = _aggregate(DataOffsets.LOG_SUM, this);
-  var log_sum_sq  = _aggregate(DataOffsets.LOG_SUM_SQ, this);
+  var log_sum     = this.precomputeAggregateQuantity(Telemetry.DataOffsets.LOG_SUM);
+  var log_sum_sq  = this.precomputeAggregateQuantity(Telemetry.DataOffsets.LOG_SUM_SQ);
```
Comment (vdjeric) [inline v1/telemetry.js]: what is log sum squares used for?

Comment (Uberi) [inline v1/telemetry.js]: We use this to calculate the geometric means, as part of `Telemetry.Histogram.geometricMean()`.

**Comment by Uberi** [comment]
Merged `chart-library` branch, which had various additional fixes and tweaks from the ui-fixes-2 branch.

The new version is now live at http://police-heel-16682.bitballoon.com/ (different from previous site).

**Comment by vdjeric** [comment]
The filter options for the default histogram aren't loaded :) I'm going to merge your chart.js pull request, can you rebase this pull on top of it?

**Comment by Uberi** [comment]
Ah, that was already fixed in the chart-library branch (I had to clear cookies first, however).

I've merged all the changes to master, and this branch is now live at http://anthony-zhang.me/telemetry-dashboard/.

File: src/dashboard.js [hunk 3]
```diff
@@ -735,42 +737,43 @@ function renderHistogramGraph(hgram) {
...
+      barDatasetSpacing : 0,
+      barShowStroke: false,
+      scaleLabel: function(valuesObject) { return fmt(valuesObject.value); },
+      tooltipFontSize: 10,
+      tooltipTemplate: function(valuesObject) { return tooltipLabels[valuesObject.label] || valuesObject.label; },
+    });
+    
+    // Assign random colors to make it easy to differentiate between bars
+    gCurrentHistogramPlot.datasets[0].bars.forEach(function(bar) {
+      bar.fillColor = "hsla(" + Math.floor(Math.random() * 256) + ", 80%, 70%, 0.8)";
```
Comment (vdjeric) [inline src/dashboard.js]: It would be nice to always use the same colors instead of random every time

File: v1/telemetry-worker.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
+importScripts("telemetry.js", "big.js")
+
+var gActions = {
+  Histogram_count: function() {
```
Comment (vdjeric) [inline v1/telemetry-worker.js]: Just out of curiosity, have you time how much main-thread time is saved from moving these operations to a worker? if not, could you add some console logging code to time these operations and let me know what kinds of numbers you see

**Comment by vdjeric** [comment]
Looks good, merging it. Let me know the timings + fix the random colors enhancements in another pull request

**Pull request merged**